### DISCUSSION TITLE
Ship Systems - External Sensors

### DISCUSSION THREAD

**Discussion by Froginator45** [Thorium-Sim/thorium-nova#235] [Feature Proposals]
_For more information about the ship systems discussions, see https://github.com/Thorium-Sim/thorium-nova/discussions/224_

# External Sensors
For man-made objects to either fly or remain in geostationary orbit around the Earth, they need to monitor the local and remote environments (immediately around the spacecraft, local planets/objects, and as far out as the visible stars/astronomical phenomena). Many sensors on these objects are used to determine environments that they are not directly involved in, such as, the atmospheric conditions on the local planet.

Monitoring can take the form of:
- The position of the space object
- Determining whether there are any dangers to the spacecraft/object, such as solar flares (and rising temperatures)
- The possibility of another object (such as a meteorite) hitting the object in space

### Remote Sensing
A lot of these types of sensors are involved with positional changes of the spacecraft and detecting remote objects in the distance, rather than their localized environment. There are two general classes of remote sensors that are used in space environments—these are active and passive remote sensors.

#### Active
Active sensors provide their own energy to detect and illuminate any observable objects in the distance by emitting radiation towards the target of interest and measuring the reflected/backscattered light from the object in the distance. Some examples of active remote sensors include:
- Laser Altimeters
- LiDAR
- RADAR
- Scatterometers
- Sounder Instruments

#### Passive
Passive sensors, on the other hand, detect radiation, or other energy forms, that have been emitted naturally by the object—with the most common energy type being the reflection of sun rays—and this enables the object to be detected. Some examples of passive remote sensors include:
- Imaging Radiometers
- Hyperspectral Radiometers
- Spectrometers
  - Gamma Ray Spectrometer
  - UV Spectrometer
  - Vis Spectrometer
- Spectroradiometers
- Star Trackers/Scanners
  - One of the most important sensors when determining position in space.
- Sun Sensors/Tracker
  - Light entering a thin slit on top of a rectangular chamber that casts an image of a thin line on the bottom of the chamber, which is lined with a network of light-sensitive cells.

### Temperature
Temperature gradients can be much more extreme in space than they are on Earth. They can also experience very quick spikes due to solar flares and other phenomena.

There are many different types of sensors that meet these requirements which are employed in space:
- Laser communication terminals
- Thermistor sensors
- Thermocouples
- Thermopiles
- Thin film sensors
- RTD sensors

### Magnetic
Used to measure various magnetic fields (planets and other space bodies). Mostly for scientific studies.

Types of magnetic sensors that can be used for space applications include:
- Fluxgate magnetometers
- Magnetic encoders
- Angular and position sensors
- Gradiometers

### Pressure/Strain Gauges
Used to help determine hull integrity during landing and take-off. 

### Accelerometers/Gyroscopes/Gravimeters
Helpful when verifying the ships current rotation and acceleration vectors. Gravimeters can also be helpful when measuring gravity.

### Vibration
Vibration sensors are very helpful when determining the current efficiency of equipment. 

### Cameras
- Visible Spectrum
- IR
- UV
- X-ray

### Neutrino Detector
Useful normally in scientific situations. 

### Life Signs Detector
Mostly a science fiction device but definitely needed in Thorium.

## Thorium
I don't see most Thorium sensors being so granularly defined as outlined in this discussion. If anything, sensors would be lumped together in the varying categories and have the following effects (specific sensor names could be possibly randomized within damage control systems reports for increased realism without introducing too much difficulty): 
- Remote Sensing
  - The most useful sensors when related to the sensors screen. Lower efficiency might not show as much details about sensor contacts. At lower efficiency, it also might sporadically have contacts "blib" in and out. 
- Temperature
  - Not entirely sure how this would be specifically used in Thorium other than in relation to specific mission functions (detecting some  hot or cold phenomena). Maybe effects object detection efficiency.
- Magnetic
  - Not entirely sure how this would be specifically used in Thorium other than in relation to specific mission functions (detecting some magnetic phenomena). Maybe effects object detection efficiency.
- Pressure/Strain Gauges
  - Directly effects how well we know the hull integrity.
- Accelerometers/Gyroscopes/Gravimeters
  - Used for the same purpose as they are in real life - Positioning.
- Vibration
  - May determine how well we know different systems current damage levels. With damaged vibration sensors, the crew might also be less efficient at auto-fixing various equipment and therefore need more micro-managing.
- Cameras
  - As harder code to implement, the various view screens and player screens could be viewed with different camera types. Essentially, various color masks would be applied to the different camera views. Some of these color masks might also reveal various hidden items. Different layers might also be created for specific planets/space phenomena that would reveal different info to the players when viewed with different cameras.
  - When cameras are down, this might also effect some positioning and object recognition (smart camera functions).
- Neutrino Detector
  - Can be used to detect nuclear weapons aboard enemy vessels. Can also give part of the picture when trying to determine which faction a ship belongs to.
- Life Signs Detector
  - Efficiency of the life signs detector might effect how good the life signs detectors are (within and outside the ship). Lower efficiency might not show the specific species. At lower efficiency, it also might sporadically have life signs "blib" in and out. 

---
Am I missing anything? Comments, suggestions, feedback, or alternate proposals welcome!

**Comment by alexanderson1993** [comment]
> Temperature

Infrared sensors have been used in my previous controls as a way to detect cloaked ships.

> Accelerometers/Gyroscopes/Gravimeters

In consider these connected to the Thrusters system, since they work together in a feedback loop - detect the orientation of the ship, and change the orientation.

> Cameras

This is directly connected to the view screen functionality. Different kinds of cameras would be awesome to implement, but I have no idea how you would go about making the different visual layers. We can explore this after 1.0.

**Comment by alexanderson1993** [comment]
Speaking from a first-principles point of view, the purpose of the External Sensors is to tell the crew what is going on outside of their ship. This could happen in a number of ways.

### Thorium Classic
In Thorium Classic, there are four ways to see what's going on outside the ship:
- The sensor grid, a circular radar that shows relative position of objects around the crew's ship.
- Automated sensor data that is manually sent by the Flight Director
- Sensor scans, with responses provided by the Flight Director
- Probe queries, with responses provided by the Flight Director

I suppose you could also count the Targeting screen as showing what's outside the ship, since it allows you to see objects in your targeting range.

### Thorium Nova

There will definitely be more screens showing the outside of the ship, like the Navigation screen (showing the permanent objects in a solar system), the pilot screen (showing the immediate surroundings), the targeting screen, the viewscreen, - maybe even more. However, there should still be a sensors/science officer who can use the sensors to get even more data than their peers on the bridge. If their "eyes" were the same as Thorium Classic, it would be something like this:

- A grid of some kind (probably 3D) which shows the surroundings of the ship with the ability to select  objects to get more information.
- An ability to enter queries of some kind to get specific information about surrounding space and nearby objects
- A passive system that automatically provides useful information about the surrounding space (either automatically generated by the system, or pre-defined within the mission and sent based on a trigger)

I'm not saying that's how it has to be - there are probably much better, more engaging ways to fulfill our purposes than mimicking Thorium Classic (although it's a good default/first step of iteration)

Probes are their own thing altogether and should be treated in their own discussion. They're basically like little ships that can be pre-configured.

---

What I would like to see is how we can take everything in @Froginator45's proposal and map it to specific controls and user experiences for the crew. How do they access these specific kinds of sensors? What data does it provide? How does that help them? How does the sensors/science officer make a difference, especially when other crew members have views into the outside?

If you were the sensors officer, how would you want to use the power at your fingertips to help your crew succeed? I have a few ideas from my experience with Thorium Classic:
- Scan for weaknesses and capabilities on enemy ships
- Scan for the kind of radiation emitted by a nebula so you can tune your shields to the correct frequency
- Scan for power signatures within an asteroid field to find a missing ship

I would love to hear more from everyone else about how to make sensors awesome - the most coveted station on the bridge.

**Reply by Froginator45** [reply]
I definitely see sensors: 
- Helping with sensor related tasks on any damage report
- Scanning objects to reveal extra info (with varying degrees of success)
  - Life signs
    - Number
    - Race
    - Health Stats
  - Planet (and some for other astronomical bodies) stats
    - Temperature average
    - Length of day
    - Weather averages
    - Diameter
    - Density
    - Mass
    - Gravity
    - Rotation Period
    - Surface pressure
    - Magnetic field
    - Presence of water
    - Atmosphere
  - Ship stats
    - Defense - Shield strength
    - Health - Hull integrity
    - Attack - Weapons ability (tune shields to corresponding frequency)
    - Speed - Engines
    - Intelligence & Charm - Race (lol)
    - Power capabilities (Batteries/Reactor)
    - Stealth - Cloaking abilities
    - ECM/ECCM currently in use
- Scanning to better figure out position in space (if there were fog of war, this could be used to reveal areas on the map and more time/probes could help reveal more map space)
- Deploying sensor grids for enhanced information (very useful in a larger battle but more practical in a defense situation)
- Micro-asteroid (or other object) warnings might appear for sensors and they need to determine if they really are asteroids and then can mark them for the automated defense grid? A stretch maybe but would definitely add more active duties.
- Can create different sensors for use by probes or ship upgrades
  - Range upgrade
  - Sensors+ upgrade (adds gravity, chemical, and/or light sensors)
  - Memory upgrade (necessary if too many sensors are being added or if the range has been severely improved)

I realize a lot of these are probably sensor officer basics but I figured I'd cover them anyway.

**Reply by alexanderson1993** [reply]
> I realize a lot of these are probably sensor officer basics but I figured I'd cover them anyway.

Better to be explicit about these things, to make sure we implement them the best way possible.

**Reply by astrolamb-gaming** [reply]
Something to consider is to what extent will AI be handling much of this data? E.g. ship stats or asteroid detection are probably things that the computer would annotate and provide the most applicable information. 

It's possilbe that there are two questions here we need to be asking:
What information will the computer automatically provide (based on the available sensors)?
and
What information must be explicitly sought by the science officer (or sensor operator(s))?

**Comment by astrolamb-gaming** [comment]
Just a note that an advanced form of [existing remote sensing technology](https://www.grainews.ca/features/how-to-use-remote-sensing-for-crops-to-its-fullest-potential/) could correlate to a life-signs detector. 
Perhaps different types of life (e.g. plant vs human vs various aliens) could be detected using different spectra? But we don't necessarily need to make it that complex.
